### PR TITLE
Generalize ler_sqrt

### PR DESCRIPTION
**Pull request by proux01**
##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)
- ~[ ] added corresponding documentation in the headers~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

#### Overlays (to be merged before the PR)

- https://github.com/coq-community/apery/pull/15
- https://github.com/math-comp/analysis/pull/1109

### CODE REVIEW

**Comment by proux01** [comment]
No, there seems to be no Nix derivation for apery. Should I add one based on the OPAM package (is it up to date ?).

**Comment by proux01** [comment]
@pi8027 I added algebra-tactics and apery in the CI and apery is indeed the only thing to break. Would you welcome an overlay that make apery depend on MC >= 1.17 or do you need backward compatibility?

**Comment by proux01** [comment]
@pi8027 gentle ping, what would you like for Apery?

**Review by affeldt-aist — CHANGES_REQUESTED** [review]
Maybe add the generalization of `ler_psqrt` to the changelog?

**Comment by proux01** [comment]
@affeldt-aist thanks, I'm waiting for https://github.com/math-comp/algebra-tactics/pull/71 so that Apery can be ported to MC2 to complete this.

**Comment by proux01** [comment]
Thanks!

**Comment by affeldt-aist** [comment]
This just needs to be rebased.

File: mathcomp/algebra/ssrnum.v [hunk 1]
```diff
@@ -4386,23 +4386,24 @@ rewrite -(@ler_pexpn2r R 2) ?nnegrE ?sqrtr_ge0 //.
 by rewrite !sqr_sqrtr // (le_trans pa).
 Qed.
 
-Lemma ler_psqrt : {in @pos R &, {mono sqrt : a b / a <= b}}.
+Lemma ler_psqrt : {in @nneg R &, {mono sqrt : a b / a <= b}}.
```
[the remark this answers is not in the document]
Comment (proux01) [reply mathcomp/algebra/ssrnum.v:4389]: @CohenCyril according to https://github.com/math-comp/math-comp/blob/c2bc27b04699320c18f4f46e7e39a77b02be2e20/mathcomp/algebra/ssrnum.v#L69-L72 the correct prefix here would be `p` and should have been `sp` before?

**Comment by proux01** [comment]
Let's finally merge that one.

**Pull request merged**